src/attachment.ts

```diff
@@ -1,6 +1,11 @@
 export default class Attachment {
   file: File
   directory: string | undefined
+  state: 'pending' | 'saving' | 'saved'
+  id: string | null | undefined
```

Comment:
Is `undefined` necessary?

Reply:
It allows the keys to be missing.

```js
attachment.saved({})
```

```
src/attachment.ts:59:5 - error TS2322: Type 'string | null | undefined' is not assignable to type 'string | null'.
  Type 'undefined' is not assignable to type 'string | null'.

59     this.id = attributes.id
```

Reply:
I [removed the undefined values](https://github.com/github/file-attachment-element/pull/1/files/6aa6080e936d3c1f06d0a5e6f32196b86f8c13f5..e0ff6a1386509c295e60257dbebc86951744465b) with the `?.` and `??` operators.